perceval/backends/core/launchpad.py

```diff
@@ -0,0 +1,617 @@
+
+    def __fetch_issue_messages(self, issue_id):
+        """Get messages of an issue"""
+
+        for messages_raw in self.client.issue_collection(issue_id, "messages"):
+            messages = json.loads(messages_raw)
+            self._push_cache_queue('{MESSAGES}')
+            self._push_cache_queue(messages_raw)
+
+            for msg in messages['entries']:
+                msg['owner_data'] = self.__fetch_user_data('{OWNER}', msg['owner_link'])
+                yield msg
+
+    def __fetch_issue_activities(self, issue_id):
+        """Get activities on an issue"""
+
+        for activities_raw in self.client.issue_collection(issue_id, "activity"):
+            activities = json.loads(activities_raw)
+            self._push_cache_queue('{ACTIVITIES}')
+            self._push_cache_queue(activities_raw)
+
+            for act in activities['entries']:
+                act['person_data'] = self.__fetch_user_data('{PERSON}', act['person_link'])
+                yield act
+
+    def __fetch_user_data(self, tag_type, user_link):
+        """Get data associated to an user"""
+
+        user_name = self.client.user_name(user_link)
+        self._push_cache_queue(tag_type)
+        self._push_cache_queue('{USER}')
+
+        user = {}
+
+        if not user_name:
+            self._push_cache_queue('{}')
+            return user
+
+        user_raw = self.client.user(user_name)
+        self._push_cache_queue(user_raw)
+
+        user = json.loads(user_raw)
+
+        return user
+
+
+class LaunchpadClient:
+    """Client for retrieving information from Launchpad API"""
+
+    # Max retries for handled HTTP errors
+    MAX_RETRIES = 5
+    _users = {}
+
+    def __init__(self, distribution, package=None,
+                 consumer_key=None, api_token=None,
+                 items_per_page=ITEMS_PER_PAGE, sleep_time=SLEEP_TIME):
+        self.consumer_key = consumer_key
+        self.api_token = api_token
+        self.distribution = distribution
+        self.package = package
+        self.items_per_page = items_per_page
+        self.sleep_time = sleep_time
+
+    def issues(self, start=None):
+        """Get the issues from pagination"""
+
+        payload = self.__build_payload(size=self.items_per_page, operation=True, startdate=start)
+        path = self.__get_url_project()
+        return self.__fetch_items(path=path, payload=payload)
+
+    def user(self, user_name):
+        """Get the user data by URL"""
+
+        user = None
+
+        if user_name in self._users:
+            return self._users[user_name]
+
+        url_user = self.__get_url("~" + user_name)
+
+        logging.info("Getting info for %s" % (url_user))
+
+        try:
+            raw_user = self.__send_request(url_user, headers=self.__get_headers())
+            user = raw_user
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 410:
+                logger.warning("Data is not available due to HTTP 410 Gone", url_user)
```

Comment:
This raises an error because you have do add a placeholder in the string for `url_user`.
```
 logger.warning("Data is not available due to HTTP 410 Gone - %s", url_user)
```